src/Features/CSharp/Portable/InvokeDelegateWithConditionalAccess/InvokeDelegateWithConditionalAccessAnalyzer.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.InvokeDelegateWithConditionalAccess
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class InvokeDelegateWithConditionalAccessAnalyzer : DiagnosticAnalyzer
+    {
+        private static readonly DiagnosticDescriptor descriptor = new DiagnosticDescriptor(
+            IDEDiagnosticIds.InvokeDelegateWithConditionalAccessId,
+            CSharpFeaturesResources.DelegateInvocationCanBeSimplified,
+            CSharpFeaturesResources.DelegateInvocationCanBeSimplified,
+            DiagnosticCategory.Style,
+            DiagnosticSeverity.Hidden,
+            isEnabledByDefault: true,
+            customTags: DiagnosticCustomTags.Unnecessary);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(descriptor);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(SyntaxNodeAction, SyntaxKind.IfStatement);
+        }
+
+        private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
+        {
+            // look for the form "if (a != null)" or "if (null != a)"
+            var ifStatement = (IfStatementSyntax)syntaxContext.Node;
+            if (!ifStatement.Condition.IsKind(SyntaxKind.NotEqualsExpression))
+            {
+                return;
+            }
+
+            if (ifStatement.Else != null)
+            {
+                return;
+            }
+
+            var binaryExpression = (BinaryExpressionSyntax)ifStatement.Condition;
+            if (!IsNotEqualsTest(binaryExpression.Left, binaryExpression.Right) &&
+                !IsNotEqualsTest(binaryExpression.Right, binaryExpression.Left))
+            {
+                return;
+            }
+
+            // Check for both:  "if (...) { a(); }" and "if (...) a();"
+            var statement = ifStatement.Statement;
+            if (statement.IsKind(SyntaxKind.Block))
+            {
+                var block = (BlockSyntax)statement;
+                if (block.Statements.Count != 1)
+                {
+                    return;
+                }
+
+                statement = block.Statements[0];
+            }
+
+            if (!statement.Parent.IsKind(SyntaxKind.Block))
+            {
+                return;
```

Comment:
This was supposed to be checking that "ifStatement" is parented by a block . Whoops.